src/foam/nanos/cron/CronScheduler.java

```diff
@@ -55,30 +67,34 @@ public void run() {
 
     try {
       while ( true ) {
-        Date now = new Date();
-
-        cronDAO_.where(
-          MLang.AND(
-            MLang.LTE(Cron.SCHEDULED_TIME, now),
-            MLang.EQ(Cron.ENABLED, true)
-          )
-        ).select(new AbstractSink() {
-          @Override
-          public void put(Object obj, Detachable sub) {
-            Cron cron = (Cron) ((FObject) obj).fclone();
-
-            PM pm = new PM(CronScheduler.class, "cron:" + cron.getId());
-            try {
-              cron.runScript(CronScheduler.this.getX());
-            } catch (Throwable t) {
-              logger.error(this.getClass(), "Error running Cron Job", cron.getId(), t.getMessage(), t);
-            } finally {
-              pm.log(getX());
-            }
-            cronDAO_.put((FObject) cron);
-          }
-        });
+        if ( getEnabled() ) {
+          Date now = new Date();
 
+          cronDAO_.where(
+                         MLang.AND(
+                                   MLang.LTE(Cron.SCHEDULED_TIME, now),
+                                   MLang.EQ(Cron.ENABLED, true),
+                                   MLang.OR(
+                                            MLang.EQ(Cron.STATUS, ScriptStatus.UNSCHEDULED),
+                                            MLang.EQ(Cron.STATUS, ScriptStatus.ERROR)
+                                            )
+                                   )
+                         ).select(new AbstractSink() {
```

Comment:
Should there also be an orderBy() so they're run in the scheduled order?

Reply:
added